scrapy/utils/serialize.py

```diff
@@ -14,6 +16,11 @@ class ScrapyJSONEncoder(json.JSONEncoder):
     TIME_FORMAT = "%H:%M:%S"
 
     def default(self, o: Any) -> Any:
+        warnings.warn(
+            "'ScrapyJSONEncoder' is a deprecated setting.\nIt will be removed in a future version of Scrapy.",
```

Comment:
This is not `ScrapyJSONDecoder`

Reply:
i did that for uniformity as observed in other classes. I'll surely change it stacklevel=1 targetting directly the class.

Reply:
Looks like you confused two unrelated comments.

Reply:
Yes i got confused.
I think if i implement this the issue will be resolved. 
```python
def __init__(self, *args, **kwargs):
    warnings.warn(
        "'ScrapyJSONEncoder' is a deprecated setting.\nIt will be removed in a future version of Scrapy.",
        category=ScrapyDeprecationWarning,
        stacklevel=2
    )
    super().__init__(*args, **kwargs)
```
I'm asking it beforehand as I have very less experience with Oops in python.

Reply:
What I meant here is that `ScrapyJSONEncoder` and `ScrapyJSONDecoder` are different classes (they are also not settings).

My stacklevel comment was about the other places where you care calling `warn()` with the default value of it.

Reply:
(as for the implementation, putting this in `__init__` is correct)

Reply:
> What I meant here is that `ScrapyJSONEncoder` and `ScrapyJSONDecoder` are different classes (they are also not settings).

Really sorry for such a silly mistake of targetting the wrong class. I would double-check things like these from next time.